pkg/providers/discoverers/cloud_foundry/provider.go

```diff
@@ -341,15 +341,31 @@ func (c *CloudFoundryProvider) discoverFromManifestFile(filePath string) (*Appli
 		return nil, fmt.Errorf("failed to read manifest file: %v", err)
 	}
 	var manifest cfTypes.AppManifest
-	if err := yaml.Unmarshal(data, &manifest); err != nil {
+
+	if err = yaml.Unmarshal(data, &manifest); err != nil {
 		return nil, fmt.Errorf("failed to unmarshal YAML: %v", err)
 	}
-
-	app, err := parseCFApp("", manifest)
+	// Check if the file contains a single application that does not contain a space
+	if !reflect.DeepEqual(manifest, cfTypes.AppManifest{}) {
+		app, err := parseCFApp("", manifest)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create application: %v", err)
+		}
+		return &app, nil
+	}
+	c.logger.Printf("Failed to parse Application manifest. Will attempt again using the Cloud Foundry Application manifest: %s", err)
+	var cfManifest cfTypes.CloudFoundryManifest
+	if err := yaml.Unmarshal(data, &cfManifest); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal YAML: %v", err)
+	}
+	if len(cfManifest.Applications) == 0 {
+		return nil, fmt.Errorf("no applications found in %s", filePath)
+	}
+	c.logger.Printf("Found %d applications in manifest in %s", len(cfManifest.Applications), filePath)
+	app, err := parseCFApp(cfManifest.Space, *cfManifest.Applications[0])
```

Comment:
Why `cfManifest.Applications[0]` at this point I expect to loop through all the available apps or, if `--name` flag is set, to analyzed only the specified one.

Reply:
I was trying to be consistent with the spec since this is a valid format (1 app). I didn't want to venture into extending the logic, just supporting a valid use case. But you're right, I can return all or filter with name. Will work on that